### ISSUE TITLE
Question: Is it possible to dynamically remove CLI args inside an internally loaded plugin?

### ISSUE THREAD

**Issue by symonk** [pytest-dev/pytest#6087]
All of our plugins are loaded through pytest_plugins=['plugin1', 'plugin2'] etc... inside the root conftest.py file, we would like to delete some of our CLI args if '--lf' is specified basically.

I'm toying around with pytest_load_initial_conftests() etc, but to no avail, there was another hook i found that takes in (args) but it was marked as deprecated, so don't really want to use it.  I've tried the following in the root conftest:

```python
def pytest_load_initial_conftests(args, early_config, parser):
    # assuming this is for setuptools only
    breakpoint()
```

but it never gets hit, any suggestions?

basically if we are doing a run failures run (from pytest-cache) we need to remove some of our pytest.ini CLI args (I could make changes there and flip some boolean flags etc however if we can do it simply with a hook as is, it would be ideal

```python
pytest_cmdline_preparse(config, args) # deprecated
    breakpoint()
    # cant do anything with args here, remove etc
```

**Comment by RonnyPfannschmidt** [comment]
what you ask isnt supported

what you probably can do is set the options differently in pytest_configure

**Comment by symonk** [comment]
Cheers @RonnyPfannschmidt - will toy around with that!

**Comment by RonnyPfannschmidt** [comment]
@symonk what i meant was reconfiguring the actual `config.option/ini...` details from code
invocation prams wont help against things from a init
